### PR TITLE
Pokegame redone

### PR DESCRIPTION
**Pull request by appelemac**
testing if this'll work =.=

### CODE REVIEW

**Comment by appelemac** [comment]
FINALLY YOU ************* GITHUB

**Comment by appelemac** [comment]
I'll remove the remnants in the Games.cs now

**Comment by appelemac** [comment]
Don't pull yet, bugfixing atm

**Comment by appelemac** [comment]
This should do it~

**Comment by LawlyPopz** [comment]
I can confirm this is working :) as i helped test it on mine :)

File: NadekoBot/Modules/Pokegame/Pokegame.cs [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+using NadekoBot.Classes._DataModels;
+using NadekoBot.Classes.Permissions;
+
+namespace NadekoBot.Modules.pokegame
+{
+    class Pokegame : DiscordModule
+    {
+        public override string Prefix { get; } = "poke";
+        public readonly int BASEHEALTH = 500;
+        private Dictionary<ulong, Pokestats> stats = new Dictionary<ulong, Pokestats>();
```
Comment (Kwoth) [inline NadekoBot/Modules/Pokegame/Pokegame.cs]: use concurrent dictionary

File: NadekoBot/Modules/Pokegame/Pokegame.cs [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+                    var target = e.Server.FindUsers(e.GetArg("target")).FirstOrDefault();
+                    if (target == null)
+                    {
+                        await e.Channel.SendMessage("No such person.");
+                        return;
+                    }
+                    // Checking stats first, then move
+                    //Set up the userstats
+                    Pokestats userstats;
+                    if (stats.ContainsKey(e.User.Id))
```
Comment (Kwoth) [inline NadekoBot/Modules/Pokegame/Pokegame.cs]: Don't use contains ke, use TryGetValue

File: NadekoBot/Modules/Pokegame/Pokegame.cs [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+                    }
+                    else
+                    {
+                        //not really necessary now
+                        userstats = defaultStats();
+                        stats.Add(e.User.Id, userstats);
+                    }
+                    //Check if able to move
+                    //User not able if HP < 0, has made more than 4 attacks
+                    if (userstats.HP < 0)
```
Comment (Kwoth) [inline NadekoBot/Modules/Pokegame/Pokegame.cs]: <= 0

File: NadekoBot/Modules/Pokegame/Pokegame.cs [hunk 4]
```diff
@@ -0,0 +1,419 @@
...
+            manager.CreateCommands("", cgb =>
+            {
+                cgb.AddCheck(Classes.Permissions.PermissionChecker.Instance);
+
+                commands.ForEach(cmd => cmd.Init(cgb));
+
+                cgb.CreateCommand(Prefix + "attack")
+                .Description("Attacks a target with the given move")
+                .Parameter("move", ParameterType.Required)
+                .Parameter("target", ParameterType.Required)
```
Comment (Kwoth) [inline NadekoBot/Modules/Pokegame/Pokegame.cs]: Change to unparsed, check for null and search that, to allow people to attack by name without mentioning

**Comment by Kwoth** [comment]
Public fields and all methods should be pascal case. Also you may want to move poketypes to a new folder + namespace inside poke module, which should be renamed PokemonModule

**Comment by appelemac** [comment]
F*ck, this again

**Comment by appelemac** [comment]
Now I think this test'll fail

File: NadekoBot/Modules/Pokegame/Pokegame.cs (hunk 4 above)
Comment (LawlyPopz) [inline NadekoBot/Modules/Pokegame/Pokegame.cs]: If this is done then the pokeattack 'dazzling gleam' @lawlypopzz move will break

**Pull request closed without merging**